### PR TITLE
AVRO-2649 Made argument order non-enforceable avro-tools cli

### PR DESCRIPTION
**Pull request by ravowlga123**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO-2649) issues and references them in the PR title. For example, "AVRO-2649: Argument order enforced in avro-tools cli"
  - https://issues.apache.org/jira/browse/AVRO-2649
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Comment by ravowlga123** [comment]
@RyanSkraba Calling for review.

File: lang/java/tools/src/main/java/org/apache/avro/tool/SpecificCompilerTool.java [hunk 1]
```diff
@@ -62,23 +62,23 @@ public int run(InputStream in, PrintStream out, PrintStream err, List<String> ar
 
     int arg = 0;
 
-    if ("-encoding".equals(args.get(arg))) {
+    if (args.contains("-encoding")) {
```
Comment (RyanSkraba) [inline lang/java/tools/src/main/java/org/apache/avro/tool/SpecificCompilerTool.java]: I think the checks for "-encoding" and "-templateDir" are incorrect -- the index `arg` won't be positioned correctly to assign the value.

Comment (ravowlga123) [reply lang/java/tools/src/main/java/org/apache/avro/tool/SpecificCompilerTool.java]: Hi @RyanSkraba you are correct about that and I have made changes accordingly so please do review and test it when you have time.

**Comment by RyanSkraba** [comment]
Thanks for the work!  I put on my "QA" hat to give it some tests -- I came up with the following contrived command that would normally work, but would now break:

```
avro-tools.jar compile -templateDir schema schema protocol
```

i.e. Compile Avro schemas located in the `protocol/` directory using the velocity templates in the `schema/` directory.

It's a bogus example (but it's possible).  Do you think it matters?

**Comment by ravowlga123** [comment]
Hi @RyanSkraba Thanks for QA. If the command was working before making then changes then it should matter in my opinion. 

To solve this, I will create positionindex array to save positions of each flag and get the maximum index of it then add + 1 to get schema/protocol value and after that everything should be the same.

I will create a PR soon with the changes mentioned above.

**Comment by ravowlga123** [comment]
@RyanSkraba Did you get a chance to test the code after I made the changes? It would be great if you could review it.

**Pull request merged**